### DISCUSSION TITLE
How to restore to a previous brew installation

### DISCUSSION THREAD

**Discussion by Konstrukteur** [Homebrew/discussions#3757] [Everyday usage]
I installed an app, which triggered an update of Homebrew and apparently also of some apps. Now I am unable to get some apps running, for example MySQL. I would like to revert the whole Homebrew installation to the point before the update was triggered to get my apps running again and to find out what went wrong. I can rely on TimeMachine. 

Which directories do I have to recover?

I have the following directories in /usr/local :

/usr/local/bin
/usr/local/Caskroom
/usr/local/Cellar
/usr/local/etc
/usr/local/Frameworks
/usr/local/Homebrew
/usr/local/include
/usr/local/lib
/usr/local/opt
/usr/local/remotedesktop
/usr/local/sbin
/usr/local/share
/usr/local/var

Thank you in advance.

**Comment by SMillerDev** [comment]
Since Homebrew is only ever aiming to work with the latest dependencies, the chance this will fix your problems is small. Your system is more likely to work with the newest dependencies of Homebrew.

**Reply by Konstrukteur** [reply]
Thank you for your feedback. At the end I was able to resurrect MySQL, Apache and PHP by restoring the whole /usr/local directory from Timemachine. I do not know exactly what caused my broken system. I discovered half a year ago, that the formulas on some apps do not support Mojave any longer and my knowledge currently does not cover reimplementing Mojave support, if it even can be reimplemented. I refrained from asking, since you the maintainer put the notice in the app. Node as well as MySQL are two formulas, which do not compile any longer. So I went with a selective `brew upgrade formulaX` and this might have broken something or when I installed Flashrom last week something went wrong. I restored the whole /usr/local and not just Caskroom, Cellar and Homebrew, because I did not know if there are Symlinks in the other directories, which connect to the directories I knew for sure were Homebrew's. Now Apache, PHP and MySQL are back, but I do not know how to protect the system from here on. A `brew update` should be innocuous but I will refrain for now from any app upgrade on this machine. Any advice?
